kn/base/templatetags/base.py

```diff
@@ -25,6 +26,15 @@ def email_filter(value):
 def mark_safe_filter(value):
     return mark_safe(value)
 
+@stringfilter
+@register.filter(name='json')
+def json_filter(data):
+    return mark_safe(json.dumps(data)
+                         .replace('&', '\u0026')
+                         .replace('<', '\u003c')
+                         .replace('>', '\u003e')
+                         .replace('/', '\u002f'))
```

Comment:
Een mooiere workaround is de JSON als onveilige string behandelen en gewoon in de HTML zetten (binnen een onzichtbaar element), zodat die als alle andere strings wordt ge-escapet. In JavaScript kun je dan eenvoudig de inhoud daarvan uitlezen met `textContent` en in JSON omzetten met `JSON.parse`.

Ik heb net even naar de Gmail broncode gekeken. Daar wordt JSON rechtstreeks in een `<script>` blok gezet, en het lijkt erop dat de karakters `&<=>` (maar niet `"'/`) ge-escapet worden.

Ik zie zelf geen lek in beide implementaties. Wederom, zonder `&`, `<` en `>` is het nogal moeilijk om een `<script>` tag te beëindigen of er extra HTML in te zetten, zolang het maar valide JSON is.